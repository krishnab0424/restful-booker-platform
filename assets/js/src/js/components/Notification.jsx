import React from 'react';
import { Link } from 'react-router-dom';

export default class Notification extends React.Component {

    constructor() {
        super();
    }

    componentDidMount() {
        this.props.setCount();
    }

    render() {
        let alertIcon;

        if(this.props.count > 0){
            alertIcon = <i className="fa fa-inbox" style={{fontSize : "1.5rem"}}><span className="notification">{this.props.count}</span></i>
        } else {
            alertIcon = <i className="fa fa-inbox" style={{fontSize : "1.5rem"}}></i>
        }
        
        return <Link className="nav-link" to="/admin/messages">{alertIcon}</Link>;
    }
    
}