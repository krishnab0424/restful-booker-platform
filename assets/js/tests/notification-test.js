import React from 'react';
import Notification from '../src/js/components/Notification.jsx';

test('Notification renders plain inbox when no unread notifications', () => {

    const notificationComponent = shallow(<Notification setCount={() => {}} />);

    expect(notificationComponent).toMatchSnapshot();

});

test('Notification renders alert inbox when there are unread notifications', async () => {

    const notificationComponent = shallow(<Notification setCount={() => {}} count={34} />);

    expect(notificationComponent).toMatchSnapshot();

});
