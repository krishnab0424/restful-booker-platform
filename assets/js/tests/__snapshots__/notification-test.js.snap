// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders alert inbox when there are unread notifications 1`] = `
<Link
  className="nav-link"
  to="/admin/messages"
>
  <i
    className="fa fa-inbox"
    style={
      Object {
        "fontSize": "1.5rem",
      }
    }
  >
    <span
      className="notification"
    >
      34
    </span>
  </i>
</Link>
`;

exports[`Notification renders plain inbox when no unread notifications 1`] = `
<Link
  className="nav-link"
  to="/admin/messages"
>
  <i
    className="fa fa-inbox"
    style={
      Object {
        "fontSize": "1.5rem",
      }
    }
  />
</Link>
`;
